### PR TITLE
Add allow_blank option for rich editor settings fields

### PR DESCRIPTION
**Pull request by ghost1227**
Issue #2808

### CODE REVIEW

File: includes/admin/settings/register-settings.php [hunk 1]
```diff
@@ -1269,6 +1270,10 @@ function edd_rich_editor_callback( $args ) {
 
 	if ( isset( $edd_options[ $args['id'] ] ) ) {
 		$value = $edd_options[ $args['id'] ];
+
+		if( ! isset( $args['allow_blank'] ) && empty( $value ) ) {
```
Comment (chriscct7) [inline includes/admin/settings/register-settings.php]: This should be:
```if(  empty( $args['allow_blank'] ) && empty( $value ) ) {```

Comment (pippinsplugins) [inline includes/admin/settings/register-settings.php]: Yep, Chris is right.

Comment (ghost1227) [inline includes/admin/settings/register-settings.php]: Fixed

**Comment by cklosowski** [comment]
Works beautifully. Was using this setting to test it:
```
				array(
					'id'          => 'EDD_Send_Cart_text',
					'name'        => __( 'Defauilt Send Cart Message', 'edd-send-cart' ),
					'desc'        => __( 'The default text provided to the user when sending a cart.', 'edd-send-cart' ),
					'type'        => 'rich_editor',
					'allow_blank' => false,
					'std'         => __( "Dear", "edd-send-cart" ) . " {name},\n\n" . __( "{from} thought you might like to see what they are shopping for on {sitename}.", "edd-send-cart" ) . "\n\n{link}\n\n"
				)
```

Thanks @ghost1227

**Pull request merged**